frontend/submission/task_submission.tsx

```diff
@@ -0,0 +1,287 @@
+        const state = yield* appSelect();
+        const currentTask = state.task.currentTask;
+        const tests = extractTestsFromTask(currentTask, level);
+
+        return yield new Promise<TaskSubmissionResultPayload>(resolve => {
+            backgroundStore.dispatch({type: TaskActionTypes.TaskRunExecution, payload: {options: state.options, level, testId, tests, answer, resolve}});
+        });
+    }
+
+    *cancelBackgroundExecution() {
+        const backgroundStore = Codecast.environments['background'].store;
+        backgroundStore.dispatch({type: StepperActionTypes.StepperExit});
+    }
+
+    *gradeAnswer(parameters: PlatformTaskGradingParameters): Generator<any, PlatformTaskGradingResult, any> {
+        const {answer, maxScore, minScore} = parameters;
+        const state = yield* appSelect();
+
+        if (TaskPlatformMode.RecordingProgress === getTaskPlatformMode(state)) {
+            return {
+                score: minScore + (maxScore - minScore) * Number(answer) / recordingProgressSteps,
+                message: '',
+            }
+        }
+
+        if (TaskSubmissionEvaluateOn.Server === state.submission.executionMode) {
+            return yield* apply(this, this.gradeAnswerServer, [parameters]);
+        } else {
+            return yield* apply(this, this.gradeAnswerClient, [parameters]);
+        }
+    }
+
+    *gradeAnswerServer(parameters: PlatformTaskGradingParameters): Generator<any, PlatformTaskGradingResult, any> {
+        const {level, answer, maxScore, minScore} = parameters;
+        const state = yield* appSelect();
+
+        const randomSeed = state.platform.taskRandomSeed;
+        const newTaskToken = getTaskTokenForLevel(level, randomSeed);
+        const answerToken = getAnswerTokenForLevel(stringify(answer), level, randomSeed);
+
+        const serverSubmission: TaskSubmissionServer = {
+            evaluated: false,
+            date: new Date().toISOString(),
+            type: TaskSubmissionEvaluateOn.Server,
+        };
+        yield* put(submissionAddNewTaskSubmission(serverSubmission));
+
+        const submissionIndex = yield* appSelect(state => state.submission.taskSubmissions.length - 1);
+
+        yield* put(submissionChangePaneOpen(true));
+        yield* put(submissionChangeCurrentSubmissionId(submissionIndex));
+
+        const submissionData = yield* makeServerSubmission(answer, newTaskToken, answerToken);
+        if (!submissionData.success) {
+            yield* put(submissionUpdateTaskSubmission({id: submissionIndex, submission: {...serverSubmission, crashed: true}}));
+
+            return {score: 0};
+        }
+
+        const submissionId = submissionData.submissionId;
+
+        let submissionResult;
+        try {
+            submissionResult = yield* getServerSubmissionResults(submissionId);
+        } catch (e) {
+            yield* put(submissionUpdateTaskSubmission({id: submissionIndex, submission: {...serverSubmission, crashed: true}}));
```

Comment:
Ah yes good idea! I've changed this so that after the delay has expired, the submit button is enabled again but the submission keeps polling the server to get its results.